plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseSharedMetastoreTest.java

```diff
@@ -128,4 +134,57 @@ public void testShowSchemas()
                 showCreateIcebergWithRedirectionsSchema,
                 getExpectedIcebergCreateSchema("iceberg_with_redirections"));
     }
+
+    @Test
+    public void testTimeTravelAsOfVersionWithRedirection()
+    {
+        assertQuery("SELECT * FROM iceberg_with_redirections." + schema + ".nation", "SELECT * FROM nation");
+        long snapshot1 = getLatestSnapshotId();
+        long v1EpochMillis = getCommittedAtInEpochMilliSeconds(snapshot1);
+        assertUpdate(format("INSERT INTO iceberg_with_redirections.%s.nation VALUES(25, 'POLAND', 3, 'test 1')", schema), 1);
+        long snapshot2 = getLatestSnapshotId();
```

Comment:
You may want to add `Thread.sleep(1)` (same as in https://github.com/trinodb/trino/blob/4a0f84f7e9987a71e62a24faca4d90528e1be6e2/plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergReadVersionedTable.java#L46-L54) in order to avoid version collisions based on the same timestamp.

Reply:
It may very well happen that in the same millisecond there are two snapshots created and we can't distinguish then with the syntax `FOR TIMESTAMP AS OF ...` which snapshot we want to retrieve.